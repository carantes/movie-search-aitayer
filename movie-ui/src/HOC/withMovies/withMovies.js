import React, { Component } from "react";
import { connect } from "react-redux";
import { fetchGetMovies } from "Actions/movies";

const withMovies = WrappedComponent => {
  class With extends Component {
    render() {
      return <WrappedComponent {...this.props} />;
    }
  }

  const moviesSelector = ({ movies }) => ({
    loading: movies.loading,
    movies: movies.movies,
  });

  const moviesActions = dispatch => ({
    getMovies(keyword) {
      return dispatch(fetchGetMovies(keyword));
    },
  });

  return connect(
    moviesSelector,
    moviesActions,
  )(With);
};

export default withMovies;
