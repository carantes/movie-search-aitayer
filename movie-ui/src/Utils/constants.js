export const {
  API_URL = "",
  WAIT_INTERVAL = 300,
  TYPE_COUNT = 3,
} = process.env;

export const headers = {
  Accept: "application/json",
};
