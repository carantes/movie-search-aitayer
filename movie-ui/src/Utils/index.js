export * from "./constants";
export { fetchHelper } from "./fetchHelper";
export { default as mocks } from "./mocks";
