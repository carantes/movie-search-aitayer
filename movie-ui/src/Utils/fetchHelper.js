import { API_URL, headers } from "./constants";

export const fetchHelper = (url, options = {}) =>
  fetch(`${API_URL}/${url}`, { headers, ...options }).then(res => res.json());
