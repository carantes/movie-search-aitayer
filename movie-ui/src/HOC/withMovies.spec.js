import React from "react";
import { mount } from "enzyme";
import configureStore from "redux-mock-store";
import nock from "nock";
import withMovies from "./withMovies";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import { mocks } from "Utils";

const mockStore = configureStore([thunk]);
const initialState = {
  movies: {
    movies: [],
    loading: false,
  },
};
const store = mockStore(initialState);

describe("withMovies HOC", () => {
  let wrapped;

  beforeEach(() => {
    let dummy = () => <div />;
    const HOC = withMovies(dummy);
    wrapped = mount(
      <Provider store={store}>
        <HOC />
      </Provider>,
    );
  });

  it("should render properly", () => {
    expect(wrapped).toMatchSnapshot();
  });

  it("should pass getMovie prop to children", () => {
    expect(
      wrapped
        .children()
        .children()
        .props().getMovies,
    ).toBeInstanceOf(Function);
  });

  it("should pass an empty movies list prop to children", () => {
    expect(
      wrapped
        .children()
        .children()
        .props().movies,
    ).toEqual(initialState.movies.movies);
  });

  it("should send loading prop to children", () => {
    expect(
      wrapped
        .children()
        .children()
        .props().loading,
    ).toEqual(initialState.movies.loading);
  });

  // Actions
  it("should dispatch fetchCategories action", async () => {
    nock("http://localhost")
      .get("/api/search")
      .query({ keyword: "avengers" })
      .reply(200, mocks.movies);

    await wrapped
      .children()
      .children()
      .props()
      .getMovies("avengers");

    expect(store.getActions()).toMatchSnapshot();

    nock.cleanAll();
  });
});
