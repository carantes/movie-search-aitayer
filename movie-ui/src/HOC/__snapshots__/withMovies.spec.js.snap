// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withMovies HOC should dispatch fetchCategories action 1`] = `
Array [
  Object {
    "loading": true,
    "payload": "avengers",
    "type": "REQUEST_MOVIES",
  },
  Object {
    "keyword": "avengers",
    "loading": false,
    "movies": Array [
      Object {
        "Poster": "https://m.media-amazon.com/images/M/MV5BNDYxNjQyMjAtNTdiOS00NGYwLWFmNTAtNThmYjU5ZGI2YTI1XkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_SX300.jpg",
        "Title": "The Avengers",
        "Type": "movie",
        "Year": "2012",
        "imdbID": "tt0848228",
      },
      Object {
        "Poster": "https://m.media-amazon.com/images/M/MV5BMTM4OGJmNWMtOTM4Ni00NTE3LTg3MDItZmQxYjc4N2JhNmUxXkEyXkFqcGdeQXVyNTgzMDMzMTg@._V1_SX300.jpg",
        "Title": "Avengers: Age of Ultron",
        "Type": "movie",
        "Year": "2015",
        "imdbID": "tt2395427",
      },
      Object {
        "Poster": "https://m.media-amazon.com/images/M/MV5BMjMxNjY2MDU1OV5BMl5BanBnXkFtZTgwNzY1MTUwNTM@._V1_SX300.jpg",
        "Title": "Avengers: Infinity War",
        "Type": "movie",
        "Year": "2018",
        "imdbID": "tt4154756",
      },
      Object {
        "Poster": "https://m.media-amazon.com/images/M/MV5BMTc5MDE2ODcwNV5BMl5BanBnXkFtZTgwMzI2NzQ2NzM@._V1_SX300.jpg",
        "Title": "Avengers: Endgame",
        "Type": "movie",
        "Year": "2019",
        "imdbID": "tt4154796",
      },
    ],
    "type": "RECEIVE_MOVIES",
  },
]
`;

exports[`withMovies HOC should render properly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ConnectFunction>
    <With
      getMovies={[Function]}
      loading={false}
      movies={Array []}
    >
      <dummy
        getMovies={[Function]}
        loading={false}
        movies={Array []}
      >
        <div />
      </dummy>
    </With>
  </ConnectFunction>
</Provider>
`;
